years/2020/04/solutions.ts

```diff
@@ -0,0 +1,75 @@
+export const part1 = (input: string[]) => {
+  return input.filter(val => {
+    const passwordObj = passwordObject(val)
+
+    if (!passwordHasValidKeys(passwordObj)) {
+      return false
+    }
+
+    return true
+  }).length
+}
+
+export const part2 = (input: string[]) => {
+  return input.filter(val => {
+    const passwordObj = passwordObject(val)
+
+    if (!passwordHasValidKeys(passwordObj)) {
+      return false
+    }
+
+    return Object.keys(passwordObj).every(key => {
+      const value = passwordObj[key]
+      switch (key) {
+        case 'byr':
+          return valueBetween(value, 1920, 2002)
+        case 'iyr':
+          return valueBetween(value, 2010, 2020)
+        case 'eyr':
+          return valueBetween(value, 2020, 2030)
+        case 'hgt':
+          const measurement = value.substr(-2)
+          const length = value.slice(0, value.length - 2)
```

Comment:
You could do something like

```suggestion
          const [ length, measurement ] = [value.slice(0, -2), value.slice(-2)]
```